ddl/generated_column.go

```diff
@@ -383,7 +383,9 @@ func checkIllegalFn4Generated(name string, genType int, expr ast.ExprNode) error
 
 func checkIndexOrStored(tbl table.Table, oldCol, newCol *table.Column) error {
 	if oldCol.GeneratedExprString == newCol.GeneratedExprString {
-		return nil
+		if oldCol.FieldType.Equal(&newCol.FieldType) {
```

Comment:
Could we update the comment of `rule 5`:https://github.com/pingcap/tidb/pull/43655/files#diff-0485729b654feaebae798e651571d5fd1091fbedb374ac61518e1536de7f37c4R200?